src/Data/List/Properties.agda

```diff
@@ -673,15 +674,15 @@ module _ {P : Pred A p} (P? : Decidable P) where
 
   takeWhile++dropWhile : ∀ xs → takeWhile P? xs ++ dropWhile P? xs ≡ xs
   takeWhile++dropWhile []       = refl
-  takeWhile++dropWhile (x ∷ xs) with P? x
-  ... | yes _ = cong (x ∷_) (takeWhile++dropWhile xs)
-  ... | no  _ = refl
+  takeWhile++dropWhile (x ∷ xs) with does (P? x)
```

Comment:
Could you just confirm for me that these proofs still go through without this change? i.e. that changing the definition of `takeWhile` doesn't **require** us to change the proofs.

Reply:
Yep. When I was working on these, the file checked before and after the changes to each lemma.